core/server/data/importer/importers/data/settings.js

```diff
@@ -80,6 +80,22 @@ class SettingsImporter extends BaseImporter {
             return ['core', 'theme'].indexOf(data.type) === -1;
         });
 
+        const newIsPrivate = _.find(this.dataToImport, {key: 'is_private'});
+        const oldIsPrivate = _.find(this.existingData, {key: 'is_private'});
+
+        this.dataToImport = _.filter(this.dataToImport, (data) => {
+            return data.key !== 'is_private';
+        });
+
+        // Only show warning if we are importing a private site into a non-private site.
+        if (oldIsPrivate === false && newIsPrivate === true) {
```

Comment:
```suggestion
        if ((oldIsPrivate && newIsPrivate) && (!!oldIsPrivate.value === false && !!newIsPrivate.value === true)) {

```

The check should take into account that these are models being compared `.value` bit, and that they might not be present in the imported file and the `value` might also be a string in the incoming data :grimacing:

Reply:
P.S. might be worth to get the new behavior covered a little with unit tests if not too much extra work